src/material_library.h

```diff
@@ -0,0 +1,120 @@
+#include <stdlib.h>
+#include <iostream>
+#include <map>
+#include <string>
+#include <string>
+#include <vector>
+#include "material.h"
+
+namespace pyne {
+
+typedef std::map<std::string, pyne::Material> mat_map;
+typedef mat_map::iterator mat_iter;
```

Comment:
I like the `mat_map` typedef. `mat_iter` may be able to go with use of `auto`.

Reply:
yes but that's implies C++11 I believe, is it a standard that we are using/enforcing ?

Reply:
I like the `auto` for `map_iter`!

Reply:
Glad you like it! For clarity, I just mean that the typedef might become unused/unnecessary if auto is used the right place.

Reply:
As for C++11, I guess I assumed it was being used because auto is used elsewhere in the PR.

Reply:
yes, do we enforce c++11 in `PyNE` ? If not I should remove those `auto`s

Reply:
I don't know. A brief look at the CMake files didn't show them setting that standard, but I know most new compilers support it by default these days. I'd support the update if it isn't supported now.

Reply:
I am issuing a PR (see #1106) right now for this !

Reply:
removed the `map_iter` and switched to `auto` usaged